slasher/service/data_update.go

```diff
@@ -0,0 +1,31 @@
+package service
+
+import (
+	"time"
+
+	ptypes "github.com/gogo/protobuf/types"
+)
+
+func (s *Service) validatorUpdater() {
+	tick := time.Tick(2000 * time.Millisecond)
+	var finalizedEpoch uint64
+	for {
+		select {
+		case <-tick:
+			ch, err := s.beaconClient.GetChainHead(s.context, &ptypes.Empty{})
+			if err != nil {
+				log.Error(err)
+				break
+			}
+			if ch != nil {
+				if ch.FinalizedEpoch > finalizedEpoch {
+					log.Infof("Finalized epoch %v", ch.FinalizedEpoch)
```

Comment:
whats the point of this whole chunk of code ? if the finalized epoch increases you just log it. And this condition will always be true since the finalized epoch is zero and its values is never updated

Reply:
it is only a stub for later pr which update things every finalised epoch change

Reply:
changed the function name

Reply:
Please add a comment and mention it's a stub. A github issue to refer would be nice here

Reply:
can you also handle a cancelled context ? `<- ctx.Done`

Reply:
added handling. thanks